File: Basic/Python/Day-06/SelectionSort.py
nums = [21,34,23,54,65,45]

def SelectionSort(nums):
    n = len(nums)
    for i in range(n-1):
        minIndex = i
        for j in range(i+1,n):
            if nums[j]<nums[minIndex]:
                minIndex = j
        if minIndex != i:
            nums[i],nums[minIndex] = nums[minIndex],nums[i]

SelectionSort(nums)
print(f"List sorted by Selection Sort: {nums}")

File: BasicToAdvance/Java/Day-06/SelectionSort.java
import java.util.Arrays;

public class SelectionSort {
    public static void main(String[] args) {
        int[] arr = { 12, 15, 10, 20, 8 };

        SelectionSortAsc(arr);
        System.out.println("Array sorted using Selection Sort: " + Arrays.toString(arr));
    }

    public static void SelectionSortAsc(int[] arr) {
        int n = arr.length;
        for (int i = 0; i < n - 1; i++) {
            int minIndex = i;
            for (int j = i + 1; j < n; j++) {
                if (arr[j] < arr[minIndex]) {
                    minIndex = j;
                }
            }
            if (minIndex != i) {
                int temp = arr[i];
                arr[i] = arr[minIndex];
                arr[minIndex] = temp;
            }
        }
    }
}
